src/services/uniswap_v3.js

```diff
@@ -0,0 +1,276 @@
+import { logger } from './logger';
+
+const debug = require('debug')('router')
+const math =  require('mathjs')
+const uni = require('@uniswap/sdk')
+const ethers = require('ethers')
+const routerArtifact = require('../static/uniswap_v3_router_abi.json')
+// const routeTokens = require('../static/uniswap_route_tokens.json')
+
+// constants
+const FeeAmount = { LOW: 500, MEDIUM: 3000, HIGH: 10000 };
+const ROUTER = process.env.UNISWAP_V3_ROUTER
+const GAS_LIMIT = process.env.UNISWAP_GAS_LIMIT || 150688;
+const TTL = process.env.UNISWAP_TTL || 300;
+const UPDATE_PERIOD = process.env.UNISWAP_UPDATE_PERIOD || 300000;  // stop updating pair after 5 minutes from last request
+
+export default class UniswapV3 {
+  constructor (network = 'mainnet') {
+    this.providerUrl = process.env.ETHEREUM_RPC_URL
+    this.network = process.env.ETHEREUM_CHAIN
+    this.provider = new ethers.providers.JsonRpcProvider(this.providerUrl)
+    this.router = ROUTER;
+    this.slippage = math.fraction(process.env.UNISWAP_ALLOWED_SLIPPAGE)
+    this.allowedSlippage = new uni.Percent(this.slippage.n, (this.slippage.d * 100))
+    this.pairsCacheTime = process.env.UNISWAP_PAIRS_CACHE_TIME
+    this.gasLimit = GAS_LIMIT
+    this.expireTokenPairUpdate = UPDATE_PERIOD
+    this.zeroReserveCheckInterval = process.env.UNISWAP_NO_RESERVE_CHECK_INTERVAL
+    this.zeroReservePairs = {} // No reserve pairs
+    this.tokenList = {}
+    this.pairs = []
+    this.tokenSwapList = {}
+    this.cachedRoutes = {}
+
+    switch (network) {
+      case 'mainnet':
+        this.chainID = uni.ChainId.MAINNET;
+        break;
+      case 'kovan':
+        this.chainID = uni.ChainId.KOVAN;
+        break;
+      default:
+        const err = `Invalid network ${network}`
+        logger.error(err)
+        throw Error(err)
```

Comment:
You have a point. But the gateway shouldn't start anyway if network isn't set correctly.

Reply:
In this case better way will check configuration on starting of gateway without checking inside constructor of every module separately. But this may require additional refactoring and therefore time.